README.md

```diff
@@ -1,3 +1,94 @@
 # promises-debugging
 
-Placeholder
+Promises have been available to Node.js developers for several years in various
+forms (as a built-in feature of the language or as npm modules). However,
+debugging errors created from promises tasks can still present some challenges
+in some use cases.
+
+As a result, an _informal_ working group was formed with the goal to make the
+experience of Node.js developers less challenging when debugging such errors.
+
+## Goals
+
+The goals of this working group are:
+
+- Identifying, investigating and proposing solutions for gaps in promises error
+  debugging use cases.
+
+- Documenting best practices about debugging promises errors.
+
+## Scope
+
+Promises debugging and error handling can cover a large surface area of the
+Node.js runtime. We want to make sure this working group is focused on just the
+critical parts of that surface area so that we can make concrete progress as
+quickly as possible.
+
+Other individuals or groups can tackle the challenges that are out of scope for
+this group.
+
+### In scope
+
+- Ability to use post-mortem debugging techniques (using both OS level core
+  dumps and other forms of artifacts) to debug unhandled errors created by
+  promises tasks.
+
+- Ability to use live debugging techniques to debug unhandled errors created by
+  promises tasks.
+
+### Out of scope
+
+Any aspect of debugging promises that is not critical to _enabling_ debugging
+use cases is not in scope. For instance:
+
+- the performance of promises when using live debuggers is not relevant for this
+  working group, even if it is very important for Node.js users.
+
+- the user experience of using promises, even when it relates to providing a
```

Comment:
I'm not sure I agree - I think that is relevant to debugging.

Reply:
The idea is that we want this team to focus on the smallest number of items possible to _enable_ debugging use cases, so that we don't stretch ourselves too thin or bite more than we can chew.

Regarding this specific point: I consider that as long as live and post-mortem debugging users can get as much information on any error created from a promise task or constructor, this effort will have fulfilled its goals.

Whether promises are easy to use safely wouldn't impact the ability for those users to get as much insight into those errors as possible.

It doesn't mean that making promises easier to use safely is undesirable, just that when determining what challenges this effort needs to tackle, this is where we draw the line.

Reply:
Also, to be clear, this paragraph doesn't mean that the user experience of using promises is not relevant to debugging in general, just that it's not _what this group is focusing on_. Would you agree with that? If so I can definitely rephrase.